app/src/main/java/org/oppia/app/player/state/StateViewModel.kt

```diff
@@ -32,7 +34,14 @@ class StateViewModel @Inject constructor() : ObservableViewModel() {
 
   // TODO(#164): Add a hasPendingAnswer() that binds to the enabled state of the Submit button.
   fun getPendingAnswer(): UserAnswer {
```

Comment:
I'm confused about the changes to this--why are we checking the error handler when retrieving the pending answer? Shouldn't we instead use downstream logic to avoid calling getPendingAnswer() if there's an explicit error check required and an error is present? It's fine if someone calls getPendingError() and it crashes because the input is invalid & we never checked the error--the intent of the API, I think, is that errors should be checked prior to calling getPendingAnswer.

Reply:
i had updated with a flag that returns the hasPendingAnswerError... But i believe still we need to check the getPendingAnswerError to get this flag.

Reply:
@BenHenning I believe we may now remove isExplicitErrorCheckRequired() with the help of hasPendingAnswerError... Please do confirm.

Reply:
I think getting rid of isExplicitErrorCheckRequired makes sense, but I have some concerns with the changes here. See below.

Here's more or less how I expect error handling to work at a high level for fractions:
- User types some text
- TextWatcher detects changes and checks to see if there's a real-time error. If there is, push it upwards (e.g. probably via a new method in InteractionAnswerReceiver such as ``onAnswerRealTimeError`` that can also take a null value to indicate there is no longer an error)
- StateFragmentPresenter receives the error and displays it; in a subsequent PR, it could also disable the submit button
- If there is no real-time error or the error gets cleared, when the user tries to submit we call into the interaction handler first for a submit time error--if one is present, it's displayed and the answer is not submitted (in a subsequent PR we may also disable the submit button until they make a change, but I'm not convinced on this yet)
- If there is no submit-time error, then we submit the answer and proceed as normal

The key differences here are:
1. The handler pushes real-time errors, and needs to be queried for compile-time errors
2. Displaying the error becomes the responsibility of StateFragmentPresenter (to simplify submit button logic, and to generalize it)--the fractions input view would still own the error view, this just simplifies it
3. ``getPendingAnswer`` doesn't need any logic beyond what it already has--StateFragmentPresenter shouldn't call it unless there is an actual answer

Hopefully much of the rest of this gets a bit simpler with the above approach, too.

Reply:
Done PTAL